### PR TITLE
feat: [CHK-3108] Provide iuv as notice id for wisp deprecation

### PR DESCRIPTION
**Pull request by petretiandrea**
Depends on https://github.com/pagopa/pagopa-ecommerce-commons/pull/213

#### List of Changes
- improve close payment logging by providing transaction id, activation client and effective client
- build close payment using effective client mapping (e.g. `WISP_REDIRECT` -> `CHECKOUT_CART` to nodo)
- build email template using IUV as payment notice id when client id `WISP_REDIRECT`

#### Motivation and Context
Due to WISP dismantling we need to perform close payment using a new client id `CHECKOUT_CART` also we must outcome email using IUV as payment notice number

#### How Has This Been Tested?
On DEV by creating a WISP redirect cart

#### Screenshots (if appropriate):
Screenshot 2024-08-06 alle 15 53 45

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/services/v2/NodeService.kt [hunk 1]
```diff
@@ -172,6 +172,10 @@ class NodeService(
           .stream()
           .mapToInt { el -> el.transactionAmount.value }
           .sum())
+    logger.info(
+      "Building close payment for cancellation with effective clientId: [{}], activation client is: [{}]",
+      transactionWithCancellation.clientId?.effectiveClient?.name,
```
Comment (giovanniberti) [inline src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/services/v2/NodeService.kt:177]:
```suggestion
      transactionWithCancellation.clientId.effectiveClient.name,
```

Shouldn't `clientId` always be non-null?

**Review by giovanniberti — DISMISSED** [review]

File: src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/services/v2/NodeService.kt (hunk 1 above)
Comment (petretiandrea) [reply src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/services/v2/NodeService.kt:177]: Yes, you're right. I fix it

Comment (petretiandrea) [reply src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/services/v2/NodeService.kt]: Solved by https://github.com/pagopa/pagopa-ecommerce-event-dispatcher-service/pull/319/commits/3886a7c9799cdf26e0ecd2491b069464e530fdf8

**Review by giovanniberti — DISMISSED** [review]

**Review by giovanniberti — APPROVED** [review]

File: src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/services/v2/NodeService.kt [hunk 2]
```diff
@@ -172,6 +172,10 @@ class NodeService(
           .stream()
           .mapToInt { el -> el.transactionAmount.value }
           .sum())
+    logger.info(
+      "Building close payment for cancellation with effective clientId: [{}], activation client is: [{}]",
```
Comment (infantesimone) [inline src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/services/v2/NodeService.kt:176]: can be useful log `transactionId`

Comment (petretiandrea) [reply src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/services/v2/NodeService.kt]: Solved by https://github.com/pagopa/pagopa-ecommerce-event-dispatcher-service/pull/319/commits/3ee8c94f06b1fe5b045b5a66712d8a6f3e77fc65

**Comment by pietro-tota** [comment]
please fix tests, they are broken

**Review by giovanniberti — APPROVED** [review]

**Pull request merged**